Automattic/wp-calypso
# composite-checkout: Move contact form to ContactSlot

#### Changes proposed in this Pull Request

Previously, the contact step (step 2 of the checkout form) was provided by the payment method object. However, we determined that the fields in the contact form actually depend only on the contents of the line items and the logic of the host page, not on the payment method at all.

This change removes the `BillingContactComponent` property of payment methods and creates a new customization slot called `ContactSlot` on the `Checkout` component which will be displayed in the second step so that the host page can define its behavior.

So what happens if the `ContactSlot` is not provided? We could make it a required prop, but I can imagine a situation where the contact form is not needed. Therefore I left it optional and made the checkout form simply collapse to a two-step process (payment method and review) if that's the case.

#### Testing instructions

- Add a file called `packages/composite-checkout/demo/private.js` and inside it put the following: `export const stripeKey = 'TEST KEY HERE'`;
- Run `npm run composite-checkout-demo`.
- Verify that the form works as expected.
- Edit `demo/index.js` and remove the `ContactSlot` prop.
- Reload the page and verify that the form still works but has no second step.
